3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala

```diff
@@ -311,43 +311,34 @@ object ConversionUtils {
     URLEncoder.encode(str, enc)
 
   /**
-   * Creates a [[URI]] from a string.
+   * Parses a string to create a [[URI]].
+   * Parsing is relaxed, i.e. even if a URL is not correctly percent-encoded or not RFC 3986-compliant, it can be parsed.
    *
-   * A "relaxed" parsing can be done if `relaxed` is set to true.
-   *
-   * Few examples of URIs that would fail non-relaxed parsing:
-   * - URI having at least one % not followed by 2 hexadecimal digits
-   * - URI with more than one #
-   * - URI containing a |
-   * Such URIs would need a relaxed parsing.
+   * First, parsing is tried with [[URI.create]] and if it does not succeed,
+   * a more relaxed parsing is done with scala-uri.
+   * The reason why not using scala-uri directly is that it can encode input valid URIs.
+   * For example, with scala-uri :
+   * - http://example.com/pa+th?param=val+ue becomes http://example.com/pa%20th?param=val+ue
+   * - http://example.com/qs?param=a%20b becomes http://example.com/qs?param=a%2520b
    *
    * @param uri String containing the URI to parse.
-   * @param relaxed Specifies whether parsing should be relaxed or not.
-   *
    * @return [[Validation]] wrapping the result of the parsing:
    *         - [[Success]] with the parsed URI if there was no error or with [[None]] if the input was `null`.
    *         - [[Failure]] with the error message if something went wrong.
    */
-  def stringToUri(uri: String, relaxed: Boolean = false): Validation[String, Option[URI]] =
+  def stringToUri(uri: String): Validation[String, Option[URI]] =
     try {
       Option(uri).map(_.replaceAll(" ", "%20")).map(URI.create).success
     } catch {
-      case e: Exception =>
-        relaxed match {
-          case false =>
-            "Provided URI [%s] could not be parsed (without relaxed parsing). Error: [%s]"
+      case NonFatal(e) =>
+        try {
+          implicit val c = UriConfig(decoder = PercentDecoder(ignoreInvalidPercentEncoding = true))
+          Uri.parseOption(uri).map(_.toJavaURI).success
```

Comment:
I'd rather do:

```scala
for {
  scalaUri <- Uri.parseTry(uri).toEither
  uri <- Either.catchNonFatal(scalaUri.toJavaURI)
} yield uri
```

without the try/catch.

Also I'd remove the usual URI parsing above

Reply:
Neat! But are you referring to
1) Scala 2.12's `Try.toEither` ?
2) cats's `Either.catchNonFatal` ? (scalaz used here)

Ok I will remove java URI parsing then. (I also created this [issue](https://github.com/lemonlabsuk/scala-uri/issues/30) regarding the fact that `scala-uri` re-encodes already correctly percent-encoded URIs.)

Reply:
yes, there are equivalent with scalaz `Validation` for the second and you can do the first manually easily

Reply:
I can't have something as neat because:
1) null url needs to be taken care of before `parseTry`
2) a `String` is returned, and not a `Throwable`